learnr_bank/learnr_set2.Rmd

```diff
@@ -0,0 +1,468 @@
+  + **Hint 1:** Don't forget to define `example_array` again here.
+  + **Hint 2:** You can provide the full expression `example_array > 0` to the function `sum()`, or save that logical expression's output to a variable, and use the variable as the argument to `sum()`.
+  + Solution:  example_array <- c(5, 78, -3, 1, 6, 0)
+               sum(example_array > 0)
+               
+```{r array_q8, exercise = TRUE}
+
+```                 
+
+```{r array_q8-hint-1}
+#Don't forget to define `example_array` again here.
+```
+
+```{r array_q8-hint-2}
+#You can provide the full expression `example_array > 0` to the function `sum()`, or save that logical expression's output to a variable, and use the variable as the argument to `sum()`.
+```
+
+### Set 2: Answer Key
+
+``` {r, array_q1-answer}
+#Solution to Question 1
+array_numbers <- c(1,2,3,4,5)
+array_numbers
+```
+
+``` {r, array_q2-answer}
+#Solution to Question 2
+array_numbers2 <- 1:5
+array_numbers2
+```
+
+``` {r, array_q3-answer}
+#Solution to Question 3
+even_numbers <- c(2,4,6,8,10)
+even_numbers
+```
+
+``` {r, array_q4-answer}
+#Solution to Question 4
+even_numbers <- seq(2, 10, 2)
+even_numbers
+```
+
+``` {r, array_q5-answer}
+#Solution to Question 5
+mul3 <- seq(0, 999, 3)
+length(mul3)
+```
+
+``` {r, array_q6-answer}
+#Solution to Question 6
+example_array <- c(5, 78, -3, 11, 6, 0)
+example_array / 10
+```
+
+``` {r, array_q7-answer}
+#Solution to Question 7
+example_array <- c(5, 78, -3, 1, 6, 0)
+example_array <= 0
+```
+
+``` {r, array_q8-answer}
+#Solution to Question 8
+example_array <- c(5, 78, -3, 1, 6, 0)
+sum(example_array > 0)
+```
+
+### Set 3: Working with data frames
+
+This set of exercises will use a data frame (aka "tibble" - we will learn the distinction between these terms!) called `msleep`, which is built-in to R. This dataset contains information about some species of mammals as follows:
+
++ `name`: The common name
++ `genus`: The taxonomic genus
++ `vore`: Dietary strategy (carnivore, omnivore or herbivore)
++ `order`: The taxonomic order
++ `conservation`: The species' conservation status
++ `sleep_total`: Total amount of sleep, in hours, the species tends to experience
++ `sleep_rem`: Total amount of REM sleep, in hours, the species tends to experience
++ `sleep_cycle`: Length of sleep cycle, in hours, the species tends to experience
++ `awake`: Amount of time spent awake, in hours, the species tends to experience
++ `brainwt`: Average brain weight, in kilograms, for this species
++ `bodywt`: Average body weight, in kilograms, for this species
+
+
+First, in the code chunk below, examine the data frame using functions we have seen `head()`, `nrow()`, and `ncol()` to get an overall sense of the data frame itself. Then,  answer the following multiple choice questions about this data frame.
```

Comment:
Was there anything specific that you would like me to name the exercise chunks in set 3. I just did` r sleep_qx`.

Reply:
That seems good to me!